### ISSUE TITLE
Ensure "Community Name" updates quickly in header after change

### ISSUE THREAD

**Issue by pkfrank** [forem/forem#14129]
**Describe the bug**

If a Forem does not have a logo set, it will use some auto-generated text (originally `New Forem`).  This area is updated after the admin changes the "Community Name" via Config.

It updates quickly via `/admin`, but not as quickly in the public-facing Forem.

**To Reproduce**

1. Set up a brand new Forem
2. Set a new Community Name
3. Refresh the page via `/admin` and on the main homepage
4. Notice that the top left section is updated in Admin, but not in the main Forem

**Expected behavior**

After updating Community Name, it reflects immediately in the header.

**Screenshots**

Here is a video:
https://user-images.githubusercontent.com/9434801/124306709-64bbb800-db2c-11eb-9207-41915feba503.mp4

**Additional context**

Likely caching.  Similar in nature to #12473

**Comment by keshavbiswa** [comment]
I can work on this.

**Comment by rhymes** [comment]
Assigned to you @keshavbiswa, thanks! Let us know when/if you need assistance!

**Comment by keshavbiswa** [comment]
Not being able to reproduce this locally @pkfrank.

**Comment by rhymes** [comment]
@keshavbiswa have you enabled your caching? Run `bundle exec rails dev:cache` which turns on all remaining types of caching...

**Comment by keshavbiswa** [comment]
@rhymes I will enable and try to reproduce this then. Thanks 😄

**Comment by keshavbiswa** [comment]
Looks like a cache problem, coming from application.html layout 

```ruby
<% cache(release_adjusted_cache_key("top-html-and-config--#{user_signed_in?}")) do %>```

Not sure, what to change. Help please 😓  @rhymes

**Comment by rhymes** [comment]
Hi @keshavbiswa, thanks for finding the problem, you could use `bust_content_change_caches` where this change is to be made, similarly to

https://github.com/forem/forem/blob/871dfc4582d82c3dde2f43e00d7319b31d132a9d/app/controllers/admin/navigation_links_controller.rb#L3

This method busts the cache so that the next time that cached part will be recomputed

**Comment by keshavbiswa** [comment]
> Hi @keshavbiswa, thanks for finding the problem, you could use [`bust_content_change_caches`](https://github.com/forem/forem/blob/871dfc4582d82c3dde2f43e00d7319b31d132a9d/app/controllers/application_controller.rb#L201-L204) where this change is to be made, similarly to
> 
> https://github.com/forem/forem/blob/871dfc4582d82c3dde2f43e00d7319b31d132a9d/app/controllers/admin/navigation_links_controller.rb#L3
> 
> This method busts the cache so that the next time that cached part will be recomputed.
> 
> Let me know if this helps!

@rhymes This worked like a charm. Although there was already bust_content_change_caches inside the action. Not sure why that didn't bust the cache and after_action did? 😕

**Comment by rhymes** [comment]
@keshavbiswa it might have been called too early :)

**Comment by michael-tharrington** [comment]
Gonna open this one back up because it appears to be occurring again. See - https://www.loom.com/share/0957179d8e2d41a59828bbce3075c1d5

**Comment by keshavbiswa** [comment]
@rhymes I might need a bit of help to understand why this happening again? Busting cache in settings_controller.rb doesn't seem to fix it.

**Comment by Zhao-Andy** [comment]
Hi @keshavbiswa, sorry for the late response! I believe this is reoccurring because there's some memoization/caching happening here: https://github.com/forem/forem/blob/69b4c6c5e374dfc0a5f0e20d62a89462846c7d0f/app/helpers/application_helper.rb#L180-L182

It seems that when the name is updated, there needs to be some cache to clear. I'm not 100% sure where it is yet since I did a fairly quick investigation. I don't think this is connected to the navigation links controller file that you worked on in your previous PR. 

My reproduction steps are:
1. `rails dev:cache` to enable caching
2. Update the `community_name` according to @michael-tharrington's instructions in his video
3. Go back to the home page
4. See that the name does not change

It's been a bit flaky though, and sometimes it does change quickly enough, but other times it persists. Hope this helps!

**Comment by Zhao-Andy** [comment]
Resolved by #14736 since it removed that heading entirely. The bug still appears on the logo if someone hard refreshes, but I don't think that's a huge issue and will most likely be resolved by eventual caching. The bug does not appear when just simply landing on the page after changing so I think it's fine.
